lib/greenhouse_api/client.rb

```diff
@@ -33,18 +37,46 @@ def get_current_offer_for_application(application_id)
         http_method: :get,
         headers: headers,
         endpoint: "applications/#{application_id}/offers/current_offer",
-        params: {}
       )
+      compose_response(response)
+    end
 
-      if response.status == 200
-        Response.new(
-          body: response.body,
-          headers: response.headers,
-          status: response.status
-        )
-      else
-        response
-      end
+    sig { params(first_name: String, last_name: String, email: String, on_behalf_of_id: String, additional_args: String).returns(Response) }
```

Comment:
Should `additional_args` be a different type?

Reply:
According to the sorbet docs, it’s declared as String, but in the method body Sorbet knows that it’ll be a Hash from Symbol keys to String values
https://sorbet.org/docs/sigs#rest-parameters

Reply:
According to the sorbet docs, it’s declared as String, but in the method body Sorbet knows that it’ll be a Hash from Symbol keys to String values
https://sorbet.org/docs/sigs#rest-parameters

Reply:
Hmm but it could also be a boolean as well, so maybe this should be `additional_args: T.any(Boolean, String)`